Upgrades

RoyEJohnson opened the pull request:
for: openstax/unified#2296

## Notes
`glob` could update to latest, but that requires a later version of node. The CI doesn't seem to respect .nvmrc.
`styled-components` and `styled-icons` could update, but that requires making screenshots.

RoyEJohnson committed:
Upgrade Sentry

RoyEJohnson committed:
Lighthouse

RoyEJohnson committed:
Rangy

RoyEJohnson committed:
source-map-explorer

RoyEJohnson committed:
tslint

RoyEJohnson committed:
minor packages

RoyEJohnson committed:
uuid

RoyEJohnson committed:
yargs

RoyEJohnson committed:
workbox-*

RoyEJohnson committed:
ts-unused-exports

RoyEJohnson committed:
tiny-async-pool

Can migrate to v2, but requires a code change in three places.

TomWoodward commented on package.json:
```diff
@@ -36,20 +36,20 @@
-    "workbox-precaching": "^5.1.3",
-    "workbox-routing": "^5.1.3",
-    "workbox-strategies": "^5.1.3"
+    "uuid": "^7.0.3",
```
the change to the UUID package has been problematic for us because they changed their module format -- if it works its fine, but if you get strange issues check this

RoyEJohnson replied:
v8 and v9 didn't work, but v7 was ok.

TomWoodward reviewed (approved):
the errors/warnings in the CI output is consistent with what is happening on main, this looks good

RoyEJohnson committed:
stylelint

RoyEJohnson committed:
serve-static

RoyEJohnson committed:
semver-sort

RoyEJohnson committed:
redux, et al, mostly restricting versions

RoyEJohnson committed:
query-string

They added TypeScript definitions in 6.3, and that broke lint.

RoyEJohnson committed:
prop-types + prettier

RoyEJohnson committed:
path-to-regexp

RoyEJohnson committed:
mime

RoyEJohnson committed:
mdn-polyfills

RoyEJohnson committed:
Update caniuse-lite

RoyEJohnson committed:
less

RoyEJohnson committed:
remove md5-file packages

RoyEJohnson committed:
http-proxy-middleware, html-entities, history

RoyEJohnson committed:
glob, focus-with-polyfill, details-element-polyfill

Update .nvmrc

glob@<9

RoyEJohnson committed:
express, date-fns, date-fns-tz

RoyEJohnson committed:
color

and lock ts-utils version

RoyEJohnson committed:
@types/yargs, @types/uuid

RoyEJohnson committed:
Lock versions for styled-components and styled-icons

RoyEJohnson committed:
@craco/craco

RoyEJohnson committed:
@types/*

RoyEJohnson committed:
@formatjs/intl-pluralrules

RoyEJohnson committed:
@aws-sdk/*

aws

RoyEJohnson commented:
@TomWoodward I'm scaling the PR back to just the Dev Dependencies for now. I'm guessing that would be the safest batch of changes. Recent deployments were crashing despite successful local lint/build/test/run, so I'm hoping this will get back to successful deployment.

RoyEJohnson commented:
Apparently, when it tells you it was deployed, it takes some time before the deployment is actually functional? Things I am learning...

RoyEJohnson committed:
@babel/*

RoyEJohnson committed:
Final yarn upgrade outdated

RoyEJohnson commented:
@TomWoodward I have finished going through the outdated packages to determine what is the most recent version they can be upgraded to and still build successfully. `yarn upgrade outdated` is currently a no-op. :)
If you would prefer that I split the upgrades into multiple PRs, I can do that. Do you have a preferred grouping (e.g., dev dependencies in one, run dependencies in another)?

TomWoodward reviewed (changes requested):
please double check that the types definition versions line up with the package versions, otherwise this looks fine

RoyEJohnson committed:
match @types to package versions

remove  @types/stacktrace-js
jest
puppeteer

TomWoodward commented on package.json:
```diff
@@ -102,84 +102,82 @@
-    "@types/color": "^3.0.0",
-    "@types/express": "^4.16.0",
-    "@types/flat": "^0.0.28",
+    "@types/color": "^3.0.3",
```
the patch versions aren't going to line up, but the major/minor should 

https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library

RoyEJohnson committed:
revert typed package versions

RoyEJohnson committed:
ignore-styles

RoyEJohnson committed:
js-cookie, node-fetch

RoyEJohnson committed:
puppeteer

RoyEJohnson committed:
Fix babel-config

RoyEJohnson committed:
Match @types to package versions

Generally used ~ to constrain major+minor versions. Some packages were not matchable

RoyEJohnson committed:
sentry, jest-image-snapshot

TomWoodward commented:
re 
>  The CI doesn't seem to respect .nvmrc.

dante recently fixed this in the other repos, you could pull in the change if it helps https://github.com/openstax/assignments/pull/25/files

TomWoodward commented:
also i'm never quite sure if you're still working on this or waiting for my review, let me know when i should take a look

RoyEJohnson commented:
I think this is ready to review. I've used ~#.# for most of the @types and corresponding packages, to document the expected major-minor matching. Some packages were out of version sync already, and they may have stayed out of sync if it was the best I could do; the types package is 2 minor revisions behind in those case.
We'll leave upgrading node for another day. This is, as you noted, quite big enough as it is.

TomWoodward commented:
idk why this is working in CI but not locally, when i try to run the `yarn prerender:local` i get `Error: Cannot find module '@aws-sdk/util-base64-node'` have you seen that?

RoyEJohnson commented:
I do not have the issue with `REACT_APP_ENV=test yarn prerender:local`. It indicates success. You ran `yarn` prior?

RoyEJohnson commented:
I don't have that module installed, and I don't know what would be trying to use it.
```
% yarn why @aws-sdk/util-base64-node      
yarn why v1.22.19
[1/4] 🤔  Why do we have the module "@aws-sdk/util-base64-node"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 1.04s.
```

TomWoodward commented:
the stacktrace looks like:

```
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.js                                              
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/STSClient.js                                                  
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/STS.js                                                        
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/node_modules/@aws-sdk/client-sts/dist-cjs/index.js                                                      
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/runtimeConfig.js                                                                               
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/S3Client.js                                                                                    
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/S3.js                                                                                          
- /Users/tomwoodward/code/openstax/books-web/node_modules/@aws-sdk/client-s3/dist-cjs/index.js                                                                                       
- /Users/tomwoodward/code/openstax/books-web/script/prerender/fileUtils.ts
- /Users/tomwoodward/code/openstax/books-web/script/prerender/contentPages.tsx
- /Users/tomwoodward/code/openstax/books-web/script/prerender/local.ts
- /Users/tomwoodward/code/openstax/books-web/script/entry.js
```

i also have 
```
[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/client-s3
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/client-s3"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@aws-sdk/client-s3@3.297.0"
info Has been hoisted to "@aws-sdk/client-s3"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "6.77MB"
info Disk size with unique dependencies: "21.56MB"
info Disk size with transitive dependencies: "32.07MB"
info Number of shared dependencies: 85
✨  Done in 1.32s.

[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/util-base64-node
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/util-base64-node"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
error We couldn't find a match!
✨  Done in 1.01s.

[1602.43][upgrades][b5f224fc7]~/code/openstax/books-web: yarn why @aws-sdk/client-sts
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "@aws-sdk/client-sts"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "@aws-sdk/client-sts@3.297.0"
info Has been hoisted to "@aws-sdk/client-sts"
info Reasons this module exists
   - Hoisted from "@aws-sdk#client-cloudformation#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#client-s3#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#client-sqs#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#credential-providers#@aws-sdk#client-sts"
   - Hoisted from "@aws-sdk#credential-providers#@aws-sdk#client-cognito-identity#@aws-sdk#client-sts"
info Disk size without dependencies: "840KB"
info Disk size with unique dependencies: "11.56MB"
info Disk size with transitive dependencies: "19.81MB"
info Number of shared dependencies: 66
=> Found "@openstax/ts-utils#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
=> Found "@aws-sdk/client-dynamodb#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils#@aws-sdk#client-dynamodb" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
=> Found "@aws-sdk/client-ssm#@aws-sdk/client-sts@3.235.0"
info This module exists because "@openstax#ts-utils#@aws-sdk#client-ssm" depends on it.
info Disk size without dependencies: "812KB"
info Disk size with unique dependencies: "11.88MB"
info Disk size with transitive dependencies: "20.06MB"
info Number of shared dependencies: 67
✨  Done in 1.91s.
```

does any of that look different from you? i might have an older dependency that isn't locked or something.

staxly[bot] committed:
Merge branch 'main' into upgrades

staxly[bot] committed:
Merge branch 'main' into upgrades

staxly[bot] committed:
Merge branch 'main' into upgrades

TomWoodward commented:
i'm sure that something weird is happening here, but if i delete my node_modules and reinstall everything works. other developers may also have to do the same, but i'm tired of trying to figure it out so maybe we should just merge

staxly[bot] committed:
Merge branch 'main' into upgrades

staxly[bot] committed:
Merge branch 'main' into upgrades
